code/model/SiteTree.php

```diff
@@ -2461,6 +2485,22 @@ public function doRevertToLive() {
 		}
 		
 		$this->invokeWithExtensions('onAfterRevertToLive', $this);
+		return true;
+	}
+
+	/**
+	 * Determine if this page references a parent which is archived, and not available in stage
+	 *
+	 * @return bool True if there is an archived parent
+	 */
+	protected function isParentArchived() {
+		if($parentID = $this->ParentID) {
+			$parentPage = Versioned::get_latest_version("SiteTree", $parentID);
+			if(!$parentPage || $parentPage->IsDeletedFromStage) {
```

Comment:
I don't think this blocks merge. We don't have a clear standard one way or the other and I don't think we need to litigate this kind of thing on every change request.

@micmania1 what you might want to do is recommend that for 4.0 we update our coding standard to avoid magic properties where possible. Best for an ss-dev discussion, I think.